test/unit/notifications.server.test.ts

```diff
@@ -0,0 +1,43 @@
+import invariant from 'tiny-invariant';
+import sendNotification from '../../app/lib/notifications.server';
+
+const { NOTIFICATIONS_EMAIL_USER } = process.env;
+
+describe('Notification server lib function test', () => {
+  const recipient = 'dev@starchart.invalid';
+  const subject = 'Sample subject';
+  const text = 'Sample text';
+
+  test('sending and receiving email notification using sendNotification()', async () => {
+    const sentMailResponse = await sendNotification(recipient, subject, text);
+
+    // prevent typescript error: messageId is possibly undefined
+    invariant(sentMailResponse?.messageId, 'Expected sentMailResponse.messageId to be defined');
```

Comment:
I am confused, how does this prevent the typescript error. From what I understand, invariant is something akin to a logger?

Reply:
We were getting type-checking tests failing earlier, saying "messageId is possibly undefined," so @humphd and I discussed the issue on Slack and using the `invariant` function from `tiny-variant` throws if the value is not truthy with an error message, rendering the whole test to fail.

Reply:
yeah Im kinda confused about this as well. Why using `invariant` here works, but rather `expect`ing  `sentMailResponse.messageId` to be not null doesnt work. As far is getting rid of the ts error.

Reply:
Tested this locally, you could fix this by changing line 16 to `sentMailResponse?.messageId` instead. Then you can remove this line

Reply:
wouldn't `expect` do something similar?

Reply:
> yeah Im kinda confused about this as well. Why using `invariant` here works, but rather `expect`ing `sentMailResponse.messageId` to be not null doesnt work. As far is getting rid of the ts error.

I tried expecting `messageId` to be `typeof` string but that doesn't get rid of the error unless you put the line that initializes messageId in an if statement that checks whether `messageId` exists. I could have done that, but @humphd  recommended using `tiny-invariant` which would throw if `messageId` is undefined, rendering the test to fail.

Reply:
Was this the error you were getting? 
<img width="731" alt="image" src="https://user-images.githubusercontent.com/67077705/219122266-1a6520a2-8d7f-4d5c-95f8-f8263734eeee.png">

Reply:
> Tested this locally, I think you could fix this by changing line 16 to sentMailResponse?.messageId instead. Then you can remove this line

doing this would fix it I think

Reply:
Yeah, I am pretty sure it does. I ran typecheck after this and didn't see any errors

Reply:
Could you try this @SerpentBytes ?

> Tested this locally, I think you could fix this by changing line 16 to `sentMailResponse?.messageId` instead. Then you can remove this line